ikey/ikey.go

```diff
@@ -0,0 +1,52 @@
+package ikey
+
+import (
+	"fmt"
+
+	"github.com/apache/mynewt-artifact/sec"
+	"github.com/pkg/errors"
+)
+
+type Desc struct {
+	Private   bool
+	Algorithm string
+	PubBytes  []byte
+	Hash      []byte
+}
+
+func signKeyToDesc(key sec.PubSignKey, private bool) (Desc, error) {
+	var alg string
+	if key.Rsa != nil {
+		alg = fmt.Sprintf("RSA-%d", key.Rsa.Size()*8)
+	} else if key.Ec != nil {
+		alg = fmt.Sprintf("ECDSA-%d", key.Ec.X.BitLen())
+	} else {
+		alg = fmt.Sprintf("ED25519-%d", len(key.Ed25519)*8)
```

Comment:
This text could as well read `ED25519` because the size is fixed!

Reply:
Thanks @utzig, that is good to know.  I have fixed it as follows:
```
diff --git a/ikey/ikey.go b/ikey/ikey.go
index 54f0093..043d097 100644
--- a/ikey/ikey.go
+++ b/ikey/ikey.go
@@ -21,7 +21,7 @@ func signKeyToDesc(key sec.PubSignKey, private bool) (Desc, error) {
    } else if key.Ec != nil {
        alg = fmt.Sprintf("ECDSA-%d", key.Ec.X.BitLen())
    } else {
-       alg = fmt.Sprintf("ED25519-%d", len(key.Ed25519)*8)
+       alg = "ED25519"
    }

    pubBytes, err := key.Bytes()
```